src/core/highlight.js

```diff
@@ -45,32 +46,44 @@ export async function run(conf) {
         console.error("Timed-out waiting for highlight:", element);
         done();
       }, 4000);
-      const msg = {
-        action: "highlight",
-        code: element.textContent,
-        id: `highlight:${i}`,
-        languages: getLanguageHint(element.classList),
-      };
-      worker.addEventListener("message", function listener(ev) {
-        const {
-          data: { id, language, value },
-        } = ev;
-        if (id !== msg.id) {
-          return; // not for us!
-        }
-        element.innerHTML = value;
-        if (element.localName === "pre") {
-          element.classList.add("hljs");
-        }
-        if (language) {
-          element.classList.add(language);
-        }
-        clearTimeout(timeoutId);
-        worker.removeEventListener("message", listener);
-        done();
+      const elementsToHighlight =
+        element.querySelectorAll("code").length > 0
+          ? Array.from(element.querySelectorAll("code"))
+          : [element];
+      elementsToHighlight.map(elem => {
+        const msg = {
+          action: "highlight",
+          code: elem.innerText,
+          id: `highlight:${id++}`,
```

Comment:
Increment inside a complex expression is typically a bad thing... Would be better to do this on its own line.